### PR TITLE
GA Oracle UCP Support

### PR DESCRIPTION
**Pull request by alexsm82**
Move Oracle UCP support from Beta to GA. 

**Do not merge this PR until all feature readiness reviews are complete.**

### CODE REVIEW

**Comment by alexsm82** [comment]
#build

**Comment by alexsm82** [comment]
#build

File: dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/DataSourceService.java [hunk 1]
```diff
@@ -659,9 +659,7 @@ protected void init() throws Exception {
                 }
             }
 
-            //TODO remove noship/beta guard before GA
-            isUCP = vendorImplClassName.startsWith("oracle.ucp.jdbc.") && 
-                            (wProps.containsKey("internalDevNonshipFunctionDoNotUseProduction") || (vProps.getFactoryPID() != null && vProps.getFactoryPID().equals("com.ibm.ws.jdbc.dataSource.properties.oracle.ucp")));
+            isUCP = vendorImplClassName.startsWith("oracle.ucp");
```
Comment (aguibert) [inline dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/DataSourceService.java]: do we want to continue gating this on `<properties.oracle.ucp>`?

File: dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/internal/JDBCDriverService.java [hunk 2]
```diff
@@ -242,20 +242,16 @@ private SQLException classNotFound(Object interfaceNames, Set<String> packagesSe
         if (trace && tc.isEntryEnabled())
             Tr.entry(tc, "create", className, classloader, PropertyService.hidePasswords(props));
 
-        //TODO remove this gate before GA so this also applies to data sources configured using DataSourceDefinition
-        String vendorPropertiesPID = props instanceof PropertyService ? ((PropertyService) props).getFactoryPID() : PropertyService.FACTORY_PID;
-        if ("com.ibm.ws.jdbc.dataSource.properties.oracle.ucp".equals(vendorPropertiesPID)) {
```
Comment (aguibert) [inline dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/internal/JDBCDriverService.java]: again, wondering here why we are removing the properties PID type check

File: dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/DataSourceService.java (hunk 1 above)
Comment (alexsm82) [reply dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/DataSourceService.java]: No, the check was only there to ensure this function could only be used when the beta properties.oracle.ucp element was used.  We now want this behavior whenever UCP is used.

File: dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/internal/JDBCDriverService.java (hunk 2 above)
Comment (alexsm82) [reply dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/internal/JDBCDriverService.java]: same as above

**Pull request merged**